server/api.go

```diff
@@ -107,8 +193,21 @@ func (a *API) processActivity(w http.ResponseWriter, req *http.Request) {
 	}
 	defer req.Body.Close()
 
+	requireEncryptedContent := a.p.getConfiguration().CertificateKey != ""
 	errors := ""
 	for _, activity := range activities.Value {
+		if activity.EncryptedContent != nil {
+			content, err := a.processEncryptedContent(*activity.EncryptedContent)
+			if err != nil {
+				errors += err.Error() + "\n"
+				continue
+			}
+			activity.Content = content
+		} else if requireEncryptedContent {
```

Comment:
I don't think we should error out if there's no encrypted content. We can use the old method to fetch the message content in that case. This can be useful for backwards compatibility. For eg - If there's an issue in updating the subscriptions by the monitoring system, the change notifications that come in before the next run of the monitoring system will be discarded and that should not happen.

Reply:
In what scenario I going to receive a message without the encrypted content in a certificate based subscription?

Reply:
Suppose you have the plugin running without the certificate based subscriptions. When you update the plugin and add certificates in the config, the monitoring system attempts to delete the existing subscriptions and create new ones with certificates. But suppose there was an error in deleting the old subscription. Then we're gonna have a subscription without the certificate until the next run of the monitoring system. The monitoring system will run after one minute. In that one minute interval, we can receive change notifications which won't have the encrypted content as they are from the subscription without the certificate. So, should we discard those change notifications?

Reply:
While testing the latest build, I was also able to reproduce this scenario. After the new build is installed, the subscriptions were not recreated instantly. Instead, they were deleted and recreated after the 1st minute. During that interval, all of the change notifications were discarded.